src/api/liskApi.js

```diff
@@ -12,212 +12,179 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import { LIVE_PORT, TEST_PORT, GET, POST } from 'constants';
+import {
+	LIVE_PORT,
+	SSL_PORT,
+	TEST_PORT,
+	TESTNET_NETHASH,
+	MAINNET_NETHASH,
+} from 'constants';
 import config from '../../config.json';
-import * as privateApi from './privateApi';
-import * as utils from './utils';
+import * as resources from './resources';
+
+const commonHeaders = {
+	'Content-Type': 'application/json',
+	os: 'lisk-js-api',
+	version: '1.0.0',
+	minVersion: '>=0.5.0',
+};
 
 export default class LiskAPI {
 	constructor(providedOptions) {
 		const options = Object.assign({}, config.options, providedOptions);
+		this.ssl = options.ssl !== false;
+		this.testnet = options.testnet === true;
+		const nethash =
+			options.headers && options.headers.nethash
+				? options.headers.nethash
+				: null;
 
-		this.defaultNodes = options.nodes || config.nodes.mainnet;
-
-		this.defaultSSLNodes = this.defaultNodes;
-
-		this.defaultTestnetNodes = options.nodes || config.nodes.testnet;
+		this.defaultNodes = [...(options.nodes || config.nodes.mainnet)];
+		this.defaultTestnetNodes = [...(options.nodes || config.nodes.testnet)];
+		this.bannedNodes = [...(options.bannedNodes || [])];
 
-		this.options = options;
-		this.ssl = options.ssl;
-		this.randomNode = Boolean(options.randomNode);
-		this.testnet = options.testnet;
-		this.bannedNodes = options.bannedNodes;
-		this.node = options.node || privateApi.selectNewNode.call(this);
 		this.port =
 			options.port === '' || options.port
 				? options.port
-				: utils.getDefaultPort(options);
-		this.nethash = this.getNethash(options.nethash);
+				: this.getDefaultPort();
+		this.randomizeNodes = options.randomizeNodes !== false;
+		this.providedNode = options.node || null;
+		this.node = options.node || this.selectNewNode();
+		this.headers = this.getDefaultHeaders(nethash);
+
+		this.accounts = new resources.AccountsResource(this);
+		this.blocks = new resources.BlocksResource(this);
+		this.dapps = new resources.DappsResource(this);
+		this.delegates = new resources.DelegatesResource(this);
+		this.signatures = new resources.SignaturesResource(this);
+		this.transactions = new resources.TransactionsResource(this);
+		this.voters = new resources.VotersResource(this);
+		this.votes = new resources.VotesResource(this);
 	}
 
-	getNethash(providedNethash) {
-		const { port } = this;
-		const NetHash = this.testnet
-			? utils.netHashOptions({ port }).testnet
-			: utils.netHashOptions({ port }).mainnet;
+	get allNodes() {
+		return {
+			current: this.node,
+			default: this.defaultNodes,
+			testnet: this.defaultTestnetNodes,
+		};
+	}
 
-		if (providedNethash) {
-			NetHash.nethash = providedNethash;
-			NetHash.version = '0.0.0a';
+	get currentNodes() {
```

Comment:
Why is `currentNodes` only returning defaultNodes and not `current` ?

Reply:
if you need all, there is `allNodes`, and if you need only `current` it's `node`.
This was originally `nodes` but it collides with the resource name `nodes`, therefore, it was changed to `currentNodes`

Reply:
Okay. The name currentNodes to show defaultNodes is still confusing to me. How about currentDefaultNodes? :D

Reply:
yea I think so, too.
I think defaultNodes doesn't really make sense, it is kind of related to #588.
It shouldn't have multiple nodes stored.

Reply:
I think it is a good idea!